Source/Client/Windows/ServerBrowser.cs

```diff
@@ -114,32 +121,37 @@ private void ReloadFiles()
 
             foreach (var file in replaysDir.GetFiles("*.zip", MpVersion.IsDebug ? SearchOption.AllDirectories : SearchOption.TopDirectoryOnly).OrderByDescending(f => f.LastWriteTime))
             {
-                var replay = Replay.ForLoading(file);
-                if (!replay.LoadInfo()) continue;
-
                 var displayName = Path.ChangeExtension(file.FullName.Substring(Multiplayer.ReplaysDir.Length + 1), null);
-
-                var saveFile = new SaveFile(displayName, true, file)
+                var saveFile = new SaveFile(displayName, true, file);
```

Comment:
Pulled the displayName and saveFile assignments up to ensure they are assigned
Wrapped logic that pulls replay info from file in try/catch block to catch exceptions when reading the file.

Looks like whether or not the replay info can be read the file is still added to the list used to render the list. Is this intentional?

Reply:
After re-reading this and the other changes below this does seem intentional. Looks like the idea is that the item will still be shown in the list, but because the file couldn't be read the version will not have been set. 

What if an exception was thrown after the version was read in though? The file could still be bad but would appear valid (since the version number is all that is needed for the file to be considered valid). Should the catch block null out the version just in case?

Reply:
The probability that the version is somehow correct and the next two entries aren't is so low it's not even worth considering. In the most common case the file will be completely corrupted. It's also the way RimWorld checks the saves in its menu. Furthermore if you consider that the file can be somehow partially corrupted, the data of the save itself isn't checked at this stage in vanilla.

Reply:
Yeah, fair. Approved